setup.py

```diff
@@ -33,6 +33,6 @@ def read(file_name):
     install_requires=['boto3', 'six', 'pip'],
 
     extras_require={
-        'test': ['tox', 'flake8', 'pytest', 'pytest-cov', 'mock', 'sagemaker']
+        'test': ['tox', 'flake8', 'pytest', 'pytest-cov', 'mock', 'sagemaker', 'keras', 'tensorflow', 'numpy']
```

Comment:
shouldn't this package be agnostic of any frameworks?

Reply:
It is, but the tests need to be tested with a framework. So I picked Keras.

Reply:
I agree with you. We can add other examples that are not machine learning related.
But I would ask why not using a machine learning specific example.

Reply:
Additional dependency that we don't really need, can be slower than just a fake machine learning script, implies necessity to know keras to anyone who wants to edit this package (it's going to be open sourced after all)

Reply:
I like the idea to use a real framework in the tests. And code snippets of different frameworks in the documentation.

I agree that I will require longer to install the dependencies, but the required understand of Keras to understand the tests is extremely minimal.

I proposed us to keep the way it is for now. I am happy to change it if evidences that is not a good approach appear or more engineer agree with @nadiaya suggest approach. Please, let me know if that work for you, @nadiaya

Reply:
In addition to the points Nadia brought up, I'd also argue that including Keras distracts from the purpose of the test - the test is just for determining if the container code is interacting correctly with the user script. What if there's a bug introduced in Keras (especially since we're not freezing on a specific version of keras here) - should our tests be dependent on that?

Reply:
Keras is frozen.